File: tests/fixtures/typescript-library/src/models/book.ts
import { Genre } from './genre';

/** Maximum number of search results to return. */
export const MAX_RESULTS = 100;

/** A book in the library catalog. */
export class Book {
    constructor(
        private _title: string,
        private _isbn: string,
        private _genre: Genre,
        private _copiesAvailable: number = 1
    ) {}

    /** Get the book title. */
    title(): string {
        return this._title;
    }

    /** Get the ISBN. */
    isbn(): string {
        return this._isbn;
    }

    /** Check if the book is available for borrowing. */
    isAvailable(): boolean {
        return this._copiesAvailable > 0;
    }

    /** Get the genre. */
    genre(): Genre {
        return this._genre;
    }
}

File: tests/fixtures/java-library/src/main/java/library/models/Book.java
package library.models;

/** A book in the library catalog. */
public record Book(
    String title,
    String isbn,
    Genre genre,
    int copiesAvailable
) {
    /** Maximum number of search results to return. */
    public static final int MAX_RESULTS = 100;

    /** Compact constructor with default copies. */
    public Book(String title, String isbn, Genre genre) {
        this(title, isbn, genre, 1);
    }

    /** Check if the book is available for borrowing. */
    public boolean isAvailable() {
        return copiesAvailable > 0;
    }
}
